### DISCUSSION TITLE
Raw Data

### DISCUSSION THREAD

**Discussion by mweel1** [alpinejs/alpine#3251] [4. General]
I have some nested for each loops, and I am trying to debug it.

Is there anything equivalent to this?

`<template x-for="e in $store.base.data.entities">
            <div x-raw="e"/>
       </template>`

**Comment by SimoTod** [comment]
How about `<div x-text="JSON.stringify(e)"></div>`? 

P.S. Be careful, `<div />` is not valid html and it will just confuse Alpine. Only a few tags accept that syntax (br for example) but they are also self closing so the / is optional and rarely used nowadays

**Comment by mweel1** [comment]
Should have thought of that.  Thank you!Sent from my iPhoneOn Nov 2, 2022, at 12:40 PM, Simone Todaro ***@***.***> wrote:﻿
How about <div x-text="JSON.stringify(e)"></div>?
P.S. Be careful, <div /> is not valid html and it will just confuse Alpine. Only a few tags accept that syntax (br for example) but they are also self closing so the / is optional and rarely used nowadays
